SlotMachine.hs

```diff
@@ -295,18 +292,11 @@ testBet = result == []
 -- Main --------------------------------------------------------------------------------------------
 
 main = do
-    {
-        putStrLn "Simple Test Verification!";
-        putStrLn "=========================";
-        
-        byLine$ map (\(b, l) -> 
-            (if (b) 
-                then putStrLn ("OK. "++ l ++ " Passed!")
-                else putStrLn ("*** Test FAILED *** " ++ l)
-            )) $ zip [testNormalize, testRandom, testIndexes, testMatches, testBet] ["testNormalize", "testRandom", "testIndexes", "testMatches", "testBet"];
-
-        putStrLn "=========================";
-    }
+    putStrLn "Simple Test Verification!";
+    putStrLn "=========================";
+    mapM_ runTest $ zip [testNormalize, testRandom, testIndexes, testMatches, testBet] ["testNormalize", "testRandom", "testIndexes", "testMatches", "testBet"]
```

Comment:
`mapM` and `mapM_` are functions for automatically threading monadic actions through a list of things. The `mapM_` variant just discards the resulting list.

```hs
mapM_ :: Monad m => (a -> m b) -> [a] -> m ()
mapM_ f xs = do
    mapM f xs
    return ()

mapM :: Monad m => (a -> m b) -> [a] -> m [b]
mapM f xs = sequence (map f xs)

sequence :: Monad m => [m a] -> m [a]
sequence [] = return []
sequence (m:ms) = do
    x <- m
    xs <- sequence ms
    return (x : xs)
```